Document how to manipulate attachments via hints (#5070)
getsentry/sentry-docs: src/includes/enriching-events/add-attachment/java.mdx

To add an attachment you can either add it to the scope, pass it to any of the `capture` methods, or manipulate the list of attachments in an `EventProcessor` or `beforeSend`.

> 

Reviewer: ```suggestion
```

One linebreak is enough, fix the others too.

> ### Passing attachments to capture

Reviewer 2: We use AP style title case for headings in docs

```suggestion
### Passing Attachments to Capture
```

> You may pass attachments to any of the `capture` methods, e.g. when capturing an Exception:

Reviewer 2: We avoid latin abbreviations in docs (e.g., i.e., etc, vs)

```suggestion
You may pass attachments to any of the `capture` methods, for example, when capturing an exception:
```

```java
import io.sentry.Attachment;
import io.sentry.Hint;
import io.sentry.Sentry;

Sentry.captureException(new IllegalStateException(), Hint.withAttachment("/path/to/file.txt"))
```

```kotlin
import io.sentry.Attachment;
import io.sentry.Hint;
import io.sentry.Sentry;

Sentry.captureException(IllegalStateException(), Hint.withAttachment("/path/to/file.txt"))
```

### Adding attachments in EventProcessor

You may also manipulate attachments in `EventProcessor`:

```java
import io.sentry.Attachment;
import io.sentry.EventProcessor;
import io.sentry.Hint;
import io.sentry.Sentry;
import io.sentry.SentryEvent;
import io.sentry.protocol.Message;

public class MyClass {
  public void myMethod() {
    // Add an `EventProcessor` to the current scope. Note that
    // this helper will process *all* future events in this scope.

>     Sentry.configureScope(scope -> {

Reviewer: Maybe don't use `Sentry.configureScope` here but rather focus on the `EventProcessor` class declaration and usage.
You can mention that event processors can be added in the `Scope#addEventProcessor` and `SentryOptions#addEventProcessor`

Reviewer: Also, on Android, there's only one scope anyway.

Author: updated

        @Override
        public SentryEvent process(SentryEvent event, Hint hint) {
          hint.addAttachment(new Attachment("/path/to/file.txt"))
          return event;
        }
      });
    });

    // Send an event to Sentry. During construction of the event
    // the attachment will be added by the event processor.
    Sentry.captureMessage("Hello, world!");
  }
}
```

```kotlin
import io.sentry.Attachment;
import io.sentry.EventProcessor;
import io.sentry.Hint;
import io.sentry.Sentry
import io.sentry.SentryEvent;
import io.sentry.protocol.Message

class MyClass {
  fun myMethod() {
    // Add an `EventProcessor` to the current scope. Note that
    // this helper will process *all* future events in this scope.
    Sentry.configureScope { scope ->
      scope.addEventProcessor(object: EventProcessor {
        override fun process(event: SentryEvent, hint: Hint): SentryEvent? {
          hint.addAttachment(Attachment("/path/to/file.txt"))
          return event
        }
      })
    }

    // Send an event to Sentry. During construction of the event
    // the attachment will be added by the event processor.
    Sentry.captureMessage("Hello, world!")
  }
}
```

> ### Adding attachments in beforeSend

Reviewer 2: ```suggestion
### Adding Attachments in `beforeSend`
```

Another way of adding attachments is using `beforeSend`:

```java
import io.sentry.Attachment;
import io.sentry.Hint;
import io.sentry.Sentry;

> Sentry.init(options -> {

Reviewer: That is gonna be visible for Android too but then its `SentryAndroid`.init`, maybe we also add the `android.mdx` or skip the `Sentry.init` block and focus on the `setBeforeSend` callback declaration.

Author: updated

  options.setBeforeSend((event, hint) -> {
    hint.addAttachment(new Attachment("/path/to/file.txt"))
    return event;
  });
});
```

```kotlin
import io.sentry.Attachment;
import io.sentry.Hint;
import io.sentry.Sentry;
import io.sentry.SentryOptions.BeforeSendCallback

Sentry.init { options ->
  options.beforeSend = BeforeSendCallback { event, hint ->
    hint.addAttachment(Attachment("/path/to/file.txt"))
    event
  }
}
```

> ### Adding attachments to the scope

Reviewer 2: ```suggestion
### Adding Attachments to the Scope
```
